### PR TITLE
Add configuration params for oh-my-zsh

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

After switching to NixOS I decided to create a global ZSH profile for each of the users based on oh-my-zsh.

Unfortunately I had to set a lot of params manually using `interactiveShellInit`. This PR aims to make `oh-my-zsh` configuration easier by adding some params to be used in `configuration.nix`.

I tested it by using the following configuration in a test VM:

``` nix
{ # ...
  programs.zsh = {
    enable = true;
    oh-my-zsh = {
      enable = true;
      plugins = [ "sudo" "scala" ];
      custom = "foobar";
      theme = "norm";
    };
  };
}
```

When starting the VM (which was created by running ``NIXOS_CONFIG=`pwd`/vmtest.nix nixos-rebuild  -I nixpkgs=$HOME/Projects/nixpkgs/ build-vm``) everything worked fine.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
I would prefer to have a dedicated module and a own scope for oh my zsh (so `zsh.ohMyZshEnable` would become `zsh.oh-my-zsh.enable`).

File: nixos/modules/programs/zsh/zsh.nix [hunk 1]
```diff
@@ -143,6 +174,26 @@ in
 
         ${cfge.interactiveShellInit}
 
+        ${optionalString (cfg.enableOhMyZsh)
```
Comment (Mic92) [inline nixos/modules/programs/zsh/zsh.nix]: In an external module this would be then moved to: programs.zsh.interactiveShellInit

**Comment by Ma27** [comment]
> I would prefer to have a dedicated module and an own scope for oh my zsh

good point.

**Comment by Ma27** [comment]
@Mic92 done.
I moved everything to a new module named `oh-my-zsh.nix` and rechecked everything in a VM (see the updated description for the used configuration).
Is there anythign else to be done before merging? :-)

**Comment by Mic92** [comment]
Thanks!

**Comment by Ma27** [comment]
thanks for merging :-)

Just out of interest: the current `nixos-unstable` channel is according to the `nixpkgs-channels` repository 8 days behind. Is there anything wrong ATM (or when is a new channel update expected?)

**Pull request merged**